### PR TITLE
Bug 1175315 - Add Proxy feature for the releng API

### PR DESCRIPTION
**Pull request by gregarndt**
This makes use of the relengapi proxy that Dustin created.  It was successfully used here:
https://tools.taskcluster.net/task-inspector/#ELmePgAbRz-W_7A_RUWmSw/0

This is probably the bare minimum stuff that's needed to get a feature like this landed.  Very basic.

### CODE REVIEW

File: config/defaults.js [hunk 1]
```diff
@@ -41,6 +41,13 @@ module.exports = {
     randomizationFactor: 0.25
   },
 
+  features: {
```
Comment (gregarndt) [inline config/defaults.js]: note here: I would like to move towards grouping feature related config together.  in a future PR if this method is acceptable, I can move the images we store in the config under features:<name>:image:<image>

File: config/defaults.js [hunk 2]
```diff
@@ -41,6 +41,13 @@ module.exports = {
...
+    relengAPIProxy: {
+      image: 'quay.io/djmitche/relengapi-proxy:0.0.2',
+      token: process.env.RELENG_API_TOKEN
```
Comment (gregarndt) [inline config/defaults.js]: This is only necessary right now until secrets land and we can stuff it in there.

File: lib/features/releng_api_proxy.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      Tty: true,
+      AttachStdin: false,
+      AttachStdout: true,
+      AttachStderr: true,
+
+      Cmd: cmd
+    });
+
+    // Terrible hack to get container promise proxy.
+    this.container = docker.getContainer(this.container.id);
```
Comment (jonasfj) [inline lib/features/releng_api_proxy.js]: nit, upgrade `dockerode-promise` and this _might_ not be needed anymore...

**Comment by jonasfj** [comment]
Looks good to me :)

**Pull request merged**